File: apps/server/src/ids.ts
import { randomBytes } from 'node:crypto';

const ALPHABET = 'abcdefghijkmnpqrstuvwxyz23456789';

export function shortId(length = 8): string {
  const bytes = randomBytes(length);
  let out = '';
  for (let i = 0; i < length; i++) out += ALPHABET[bytes[i]! % ALPHABET.length];
  return out;
}

const MEMBER_COLORS = ['#e6194b', '#3cb44b', '#4363d8', '#f58231', '#911eb4', '#42d4f4', '#f032e6', '#bfef45'];
export const memberColor = (index: number): string => MEMBER_COLORS[index % MEMBER_COLORS.length]!;

File: apps/brushjam/src/brushjam/ids.py
"""Mirror of the retired Node server's src/ids.ts."""

from __future__ import annotations

import secrets

ALPHABET = "abcdefghijkmnpqrstuvwxyz23456789"

MEMBER_COLORS = [
    "#e6194b",
    "#3cb44b",
    "#4363d8",
    "#f58231",
    "#911eb4",
    "#42d4f4",
    "#f032e6",
    "#bfef45",
]


def short_id(length: int = 8) -> str:
    raw = secrets.token_bytes(length)
    return "".join(ALPHABET[b % len(ALPHABET)] for b in raw)


def member_color(index: int) -> str:
    return MEMBER_COLORS[index % len(MEMBER_COLORS)]
